demo/src/test/java/com/example/demo/controller/DeveloperControllerTest.java

```diff
@@ -57,26 +57,121 @@ public void testCreateUser() {
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
 
-        HttpEntity<Developer> requestEntity = new HttpEntity<>(user, headers);
+        HttpEntity<Developer> requestEntity = new HttpEntity<>(developer, headers);
         ResponseEntity<Developer> responseEntity = restTemplate.exchange(
                 "/users",
                 HttpMethod.POST,
                 requestEntity,
                 Developer.class
         );
 
-        //assertThat(responseEntity.getStatusCodeValue()).isEqualTo(200);
-        Developer createdUser = responseEntity.getBody();
+
+        Developer createdDeveloper = responseEntity.getBody();
 
         // Validate the response and check if the user was created
 
-        assertThat(createdUser).isNotNull();
-        assertThat(createdUser.getFirstName()).isEqualTo(user.getFirstName());
-        assertThat(createdUser.getLastName()).isEqualTo(user.getLastName());
-        assertThat(createdUser.getAge()).isEqualTo(user.getAge());
+        assertThat(createdDeveloper).isNotNull();
+        assertThat(createdDeveloper.getFirstName()).isEqualTo(createdDeveloper.getFirstName());
+        assertThat(createdDeveloper.getLastName()).isEqualTo(createdDeveloper.getLastName());
+        assertThat(createdDeveloper.getAge()).isEqualTo(createdDeveloper.getAge());
```

Comment:
This test is never going to fail because you verify it against itself

should be
```java
        assertThat(createdDeveloper).isNotNull();
        assertThat(createdDeveloper.getFirstName()).isEqualTo(developer.getFirstName());
        assertThat(createdDeveloper.getLastName()).isEqualTo(developer.getLastName());
        assertThat(createdDeveloper.getAge()).isEqualTo(developer.getAge());
```